File: src/main/java/org/bp/lab/simpleblog/service/CategoryService.java
package org.bp.lab.simpleblog.service;

import java.util.List;

import org.bp.lab.simpleblog.domain.Category;
import org.springframework.data.domain.Page;

public interface CategoryService {
	
	List<Category> findAll();
	Page<Category> getPage(int page, int size);
	Category findOne(Long id);
	Category save(Category category);
	void delete(Long id);

}

File: src/main/webapp/static/app/js/category/categoryService.js
var blog = angular.module('categoryServices', []);

blog.service('categoryService', function($http){
	
	this.url = '/api/categories';
	
	this.getAll = function(){
		return $http.get(this.url);
	};
	
	this.getPage = function(page, size){
		return $http.get(this.url, { params: { 'page': page, 'size': size }});
	};
	
	this.getOne = function(id){
		return $http.get(this.url + '/' + id);
	};
	
	this.save = function(category){
		if(category.id){
			return $http.put(this.url + '/' + category.id, category);
		}else{
			return $http.post(this.url, category);
		}
	};
	
	this.delete = function(id){
		return $http.delete(this.url + '/' + id);
	};
	
});